inc/common/purge.php

```diff
@@ -651,8 +651,18 @@ function rocket_clean_cache_theme_update( $wp_upgrader, $hook_extra ) {
  * @param array $post_data Array of unslashed post data.
  */
 function rocket_clean_post_cache_on_slug_change( $post_id, $post_data ) {
-	if ( get_post_field( 'post_name', $post_id ) !== $post_data['post_name'] ) {
-        rocket_clean_files( get_the_permalink( $post_id ) );
-    }
+	// Bail out if the post status is draft, pending or auto-draft.
+	if ( in_array( get_post_field( 'post_status', $post_id ), [ 'draft', 'pending', 'auto-draft' ], true ) ) {
```

Comment:
I'm wondering if we shouldn't switch this around to check for the `post_status` that we want instead of what we don't want. 

What do I mean?

When do we want to trigger cleaning the post cache? There are [8 major post statuses](https://wordpress.org/support/article/post-status/): `publish`, `future`, `draft`, `pending`, `private`, `trash`, `auto-draft`, `inherit`, and also custom. _The custom one is the one that worries me._ 

What do you think @crystinutzaa?

Reply:
The `custom` status is the real problem in here. My first idea was to bail out all posts which do not have `publish` post status. However I'm not sure if there could be any plugins which creates custom post statuses which are cached and should be cleaned on slug changes. So I've chosen the save way to exclude only `draft`, `auto-draft` and `pending`. Exactly as in `rocket_clean_post` function but with `pending` included:
https://github.com/wp-media/wp-rocket/blob/7723e789d158d2bf00ee5da49ef9158061e158eb/inc/common/purge.php#L81-L84

Probably there could be other exclusions for post statuses. 
Basically, this should check if the old post status was NOT published the old cache should not be cleaned.

Reply:
>However I'm not sure if there could be any plugins which creates custom post statuses which are cached and should be cleaned on slug changes.

That's an interesting and very valid point. We don't know. A custom post status could be valid for caching or it may not be. We don't know. 

Hmm, this may be a larger discussion, ie beyond the scope of this PR. Let's put a pin in this particular topic. I'll open a separate ticket where we can discuss.

Reply:
In case you need something similar someday, I made [this function](https://github.com/wp-media/imagify-plugin/blob/master/inc/functions/attachments.php#L59-L97) long ago in Imagify to get the public post statuses related to attachments. It uses [`get_post_stati()`](https://wpseek.com/function/get_post_stati/), that is handy to fetch statuses according our criteria.